metrics/src/macros.rs

```diff
@@ -0,0 +1,870 @@
+#[doc(hidden)]
+#[macro_export]
+macro_rules! metadata_var {
+    ($target:expr, $level:expr) => {{
+        static METADATA: $crate::Metadata<'static> =
+            $crate::Metadata::new($target, $level, ::core::option::Option::Some(module_path!()));
+        &METADATA
+    }};
+}
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! count {
+    () => {
+        0usize
+    };
+    ($head:tt $($tail:tt)*) => {
+        1usize + $crate::count!($($tail)*)
+    };
+}
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! key_var {
+    ($name: literal) => {{
+        static METRIC_KEY: $crate::Key = $crate::Key::from_static_name($name);
+        &METRIC_KEY
+    }};
+    ($name:expr) => {
+        $crate::Key::from_name($name)
+    };
+    ($name:literal, $($label_key:literal => $label_value:literal),*) => {{
+        static LABELS: [$crate::Label; $crate::count!($($label_key)*)] = [
+            $($crate::Label::from_static_parts(&$label_key, &$label_value)),*
```

Comment:
I don't think the label name/value have to be referenced here since they're already literals?

Reply:
28194901f7a6ed6e995248f87e8577536adc504a